src/components/Pagination/index.tsx

```diff
@@ -17,26 +18,78 @@ export class Pagination extends PureComponent<PaginationProps> {
     const handleIncrease = this.handleChange(currentPageIndex + 1);
     const handleDecrease = this.handleChange(currentPageIndex - 1);
 
-    const indexList = range(0, totalPageIndex + 1);
+    const showingNumberedIndexButtonCount = this.getValidatedShowingNumberedIndexButtonCount(
+      this.props.showingNumberedIndexButtonCount
+    );
+
+    const indexListToDisplay = this.getIndexListToDisplay(
+      currentPageIndex,
+      totalPageIndex,
+      showingNumberedIndexButtonCount
+    );
 
     return (
       <PageWrapper>
         <PageIconBtn icon={<ChevronLeft />} onClick={handleDecrease} disabled={currentPageIndex <= 0} />
-        {indexList.map((_, i) => (
+        {indexListToDisplay.map(index => (
           <PageBtn
-            color={i === currentPageIndex ? ButtonColor.ORANGE : ButtonColor.WHITE}
+            color={index === currentPageIndex ? ButtonColor.ORANGE : ButtonColor.WHITE}
             size="sm"
-            onClick={this.handleChange(i)}
-            key={`pagination-${i}`}
+            onClick={this.handleChange(index)}
+            key={`pagination-${index}`}
           >
-            {i + 1}
+            {index + 1}
           </PageBtn>
         ))}
         <PageIconBtn icon={<ChevronRight />} onClick={handleIncrease} disabled={currentPageIndex >= totalPageIndex} />
       </PageWrapper>
     );
   }
 
+  // currentPageIndex가 중앙에 보여지는 상황 때문에 홀수만 받습니다.
+  private getValidatedShowingNumberedIndexButtonCount = (showingNumberedIndexButtonCount: number | undefined) => {
+    if (!showingNumberedIndexButtonCount) {
+      return 5;
+    }
+
+    return showingNumberedIndexButtonCount % 2 === 0
+      ? showingNumberedIndexButtonCount - 1
+      : showingNumberedIndexButtonCount;
+  };
+
+  private getIndexListToDisplay = (
+    currentPageIndex: number,
+    totalPageIndex: number,
+    showingNumberedIndexButtonCount: number
+  ) => {
+    /**
+     * currentPageIndex가 중앙에서 보여질때 양옆으로 몇개까지 보여주어야 하는지 계산. 보여주어야 할 numbered
+     * index가 5개이면 양옆에는 2개씩 9개라면 4개씩
+     */
+    const aroundCurrentNumberedIndexShowingCount = Math.floor(showingNumberedIndexButtonCount / 2);
+
+    // 보여줄수 있는거 다 보여주기
+    if (totalPageIndex < showingNumberedIndexButtonCount) {
+      return range(0, totalPageIndex + 1);
+    }
+
+    // 맨 앞에서부터 보여 주어야 할 만큼 보여주기
+    if (currentPageIndex - aroundCurrentNumberedIndexShowingCount <= 0) {
```

Comment:
```suggestion
    if (currentPageIndex  <=  aroundCurrentNumberedIndexShowingCount) {
```
이렇게 하는게 이해하기 더 쉬울 듯.

Reply:
완료!